Add optional references (#43)
yiisoft/definitions: CHANGELOG.md

## 2.0.1 under development

> - Enh #43: Add `Reference::optional()` method that returns `null` on resolve dependencies when container don't have
>   entry. (@vjik)

Reviewer: ```suggestion
- Enh #43: Add `Reference::optional()` method that returns `null` when there is no dependency defined in container (@vjik)
```

Author: done
